internal/provider/image/datasource_image.go

```diff
@@ -22,9 +23,28 @@ func dataSourceImageRead(ctx context.Context, d *schema.ResourceData, meta inter
 	c := meta.(*client.Client)
 	namespace := d.Get(constants.FieldCommonNamespace).(string)
 	name := d.Get(constants.FieldCommonName).(string)
-	obj, err := c.HarvesterClient.HarvesterhciV1beta1().VirtualMachineImages(namespace).Get(ctx, name, metav1.GetOptions{})
-	if err != nil {
-		return diag.FromErr(err)
+	displayName := d.Get(constants.FieldImageDisplayName).(string)
+
+	if name != "" {
+		image, err := c.HarvesterClient.HarvesterhciV1beta1().VirtualMachineImages(namespace).Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return diag.FromErr(resourceImageImport(d, image))
+	}
+
+	if displayName != "" {
+		images, err := c.HarvesterClient.HarvesterhciV1beta1().VirtualMachineImages(namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		for i, image := range images.Items {
+			if image.Spec.DisplayName == displayName {
+				return diag.FromErr(resourceImageImport(d, &images.Items[i]))
+			}
+		}
+		return diag.FromErr(fmt.Errorf("can not find image %s in namespace %s", displayName, namespace))
```

Comment:
Note: not required for now, but would it be helpful to add the display name as a label for the quick get method?

Reply:
Need vmimage controller to add/update this label

Reply:
Yep, just considering this one as a future enhancement if it will be helpful, need to log an issue to track it, thanks.

Reply:
issue: https://github.com/harvester/harvester/issues/2630